inference-engine/src/gna_plugin/gna_device.cpp

```diff
@@ -154,15 +155,36 @@ void GNADeviceHelper::releaseModel(const uint32_t model_id) {
     checkGna2Status(status, "Gna2ModelRelease");
 }
 
+bool GNADeviceHelper::enforceLegacyCnnNeeded() const {
+    auto devVersion = getExecutionTargetDevice();
+    return isGnaLibVersion2_1 && isUpTo20HwGnaDevice(devVersion);
+}
+
+Gna2DeviceVersion GNADeviceHelper::getExecutionTargetDevice() const {
+    const volatile uint32_t executionTarget0x30 = 0x30;
+    if (executionTarget.empty()) {
+        if (detectedGnaDevVersion == Gna2DeviceVersionSoftwareEmulation)
+            return isGnaLibVersion2_1 ? static_cast<Gna2DeviceVersion>(executionTarget0x30) : Gna2DeviceVersion2_0;
+        return detectedGnaDevVersion;
+    } else if (executionTarget == InferenceEngine::GNAConfigParams::GNA_TARGET_3_0) {
+        if (!isGnaLibVersion2_1)
+            THROW_GNA_EXCEPTION << "Unsupported GNA execution target " << executionTarget << " when GNA Library version is 2.0.X.X";
+        return static_cast<Gna2DeviceVersion>(executionTarget0x30);
+    } else if (executionTarget == InferenceEngine::GNAConfigParams::GNA_TARGET_2_0) {
+        return Gna2DeviceVersion2_0;
+    }
+    THROW_GNA_EXCEPTION << "Unknown execution target: \"" << executionTarget << "\"";
+}
+
 uint32_t GNADeviceHelper::createRequestConfig(const uint32_t model_id) {
     std::unique_lock<std::mutex> lockGnaCalls{ acrossPluginsSync };
     uint32_t reqConfId;
     auto status = Gna2RequestConfigCreate(model_id, &reqConfId);
     checkGna2Status(status, "Gna2RequestConfigCreate");
-    if (gna2HwConsistency != Gna2DeviceVersionSoftwareEmulation && !isGnaLibVersion2_1) {
```

Comment:
Maybe a comment about what's happening in places like this one would be beneficial for future generations. :-)

Reply:
done